### PR TITLE
Migrate api / sdk projects to build.gradle.kts

### PR DESCRIPTION
**Pull request by anuraaga**

### CODE REVIEW

File: sdk/all/build.gradle [hunk 1]
```diff
@@ -1,47 +0,0 @@
...
-    testImplementation "org.junit-pioneer:junit-pioneer"
-}
-
-sourceSets {
-    main {
-        output.dir("build/generated/properties", builtBy: 'generateVersionResource')
-    }
-}
-
-animalsniffer {
```
Comment (Oberon00) [inline sdk/all/build.gradle:33]: What's up with the removal of this?

Comment (anuraaga) [reply sdk/all/build.gradle:33]: Was a no-op, leftover from before moving exclusions to top-level I think.

File: sdk/common/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+}
+
+sourceSets {
+    main {
+        output.dir("build/generated/properties", "builtBy" to "generateVersionResource")
+    }
+}
+
+tasks {
+    register("generateVersionResource") {
```
Comment (jkwatson) [inline sdk/common/build.gradle.kts:32]: is there a way we can not repeat this task registration? It's also up in sdk-all right now.

File: sdk/common/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        val propertiesDir = file("build/generated/properties/io/opentelemetry/sdk/common")
+        outputs.dir(propertiesDir)
+
+        doLast {
+            File(propertiesDir, "version.properties").writeText("sdk.version=${project.version}")
+        }
+    }
+}
+
+for (version in mrJarVersions) {
```
Comment (jkwatson) [inline sdk/common/build.gradle.kts:42]: this also seems like a good candidate for extraction, especially if we're going to have more MR jars in the future.

File: sdk/metrics/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+sourceSets {
+    main {
+        output.dir("build/generated/properties", "builtBy" to "generateVersionResource")
+    }
+}
+
+tasks {
+    register("generateVersionResource") {
```
Comment (jkwatson) [inline sdk/metrics/build.gradle.kts:31]: and another one here for generating resources.

File: sdk/trace/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+sourceSets {
+    main {
+        output.dir("build/generated/properties", "builtBy" to "generateVersionResource")
+    }
+}
+
+tasks {
+    register("generateVersionResource") {
```
Comment (jkwatson) [inline sdk/trace/build.gradle.kts:57]: :thonk:

**Review by jkwatson — APPROVED** [review]
Thanks!

File: sdk/trace/build.gradle.kts (hunk 5 above)
Comment (anuraaga) [reply sdk/trace/build.gradle.kts:57]: Yeah noticed the duplication we already have and want to share in another PR

**Pull request merged**